File: test-suite/src/test/java/io/micronaut/chatbots/basecamp/HelloWorldHandlerTest.java
package io.micronaut.chatbots.basecamp;

import io.micronaut.chatbots.basecamp.api.Query;
import io.micronaut.chatbots.basecamp.core.BasecampBotConfiguration;
import io.micronaut.chatbots.core.Dispatcher;
import io.micronaut.chatbots.telegram.api.Update;
import io.micronaut.chatbots.telegram.api.send.Send;
import io.micronaut.chatbots.telegram.api.send.SendMessage;
import io.micronaut.chatbots.telegram.core.TelegramBotConfiguration;
import io.micronaut.context.BeanContext;
import io.micronaut.context.annotation.Property;
import io.micronaut.json.JsonMapper;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

@MicronautTest(startApplication = false)
@Property(name = "spec.name", value = "basecamp")
class HelloWorldHandlerTest {

    @Inject
    BeanContext ctx;

    @Inject
    Dispatcher<BasecampBotConfiguration, Query, String> dispatcher;

    @Inject
    JsonMapper jsonMapper;

    @Test
    void beanOfTypeHelloWorldHandlerExists() {
        assertTrue(ctx.containsBean(HelloWorldHandler.class));
    }

    @Test
    void aboutCommandHandlerExists() throws Exception {
        String send = dispatcher.dispatch(null, jsonMapper.readValue(getClass().getResourceAsStream("/basecampAbout.json"), Query.class)).get();
        assertEquals("Bot developed with 💙 using [Micronaut](https://micronaut.io)", send.trim());
    }

    @Test
    void helloCommandHandlerExists() throws Exception {
        String send = dispatcher.dispatch(null, jsonMapper.readValue(getClass().getResourceAsStream("/basecampHello.json"), Query.class)).get();
        assertEquals("Hello World", send);
    }

    @Test
    void unknownCommandHandlerExists() throws Exception {
        String send = dispatcher.dispatch(null, jsonMapper.readValue(getClass().getResourceAsStream("/basecampText.json"), Query.class)).get();
        assertEquals("I don't know how to handle your query: some text", send);
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/chatbots/basecamp/HelloWorldHandlerTest.kt
package io.micronaut.chatbots.basecamp

import io.micronaut.chatbots.basecamp.api.Query
import io.micronaut.chatbots.basecamp.core.BasecampBotConfiguration
import io.micronaut.chatbots.core.Dispatcher
import io.micronaut.context.BeanContext
import io.micronaut.context.annotation.Property
import io.micronaut.json.JsonMapper
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

@MicronautTest(startApplication = false)
@Property(name = "spec.name", value = "basecamp")
class HelloWorldHandlerTest {

    @Inject
    lateinit var ctx: BeanContext

    @Inject
    lateinit var dispatcher: Dispatcher<BasecampBotConfiguration, Query, String>

    @Inject
    lateinit var jsonMapper: JsonMapper

    @Test
    fun beanOfTypeHelloWorldHandlerExists() {
        assertTrue(ctx.containsBean(HelloWorldHandler::class.java))
    }

    @Test
    fun aboutCommandHandlerExists() {
        val send = dispatcher.dispatch(null, jsonMapper.readValue(javaClass.getResourceAsStream("/basecampAbout.json"), Query::class.java)).get()
        assertEquals("Bot developed with 💙 using [Micronaut](https://micronaut.io)", send.trim())
    }

    @Test
    fun helloCommandHandlerExists() {
        val send = dispatcher.dispatch(null, jsonMapper.readValue(javaClass.getResourceAsStream("/basecampHello.json"), Query::class.java)).get()
        assertEquals("Hello World", send)
    }

    @Test
    fun unknownCommandHandlerExists() {
        val send = dispatcher.dispatch(null, jsonMapper.readValue(javaClass.getResourceAsStream("/basecampText.json"), Query::class.java)).get()
        assertEquals("I don't know how to handle your query: some text", send)
    }
}
